Cabal/src/Distribution/Simple/BuildToolDepends.hs

```diff
@@ -13,6 +13,26 @@ import qualified Data.Map as Map
 import           Distribution.Package
 import           Distribution.PackageDescription
 
+-- | Same as 'desugarBuildTool', but requires atomic informations (package
+-- name, executable names) instead of a whole 'PackageDescription'.
+desugarBuildToolSimple :: PackageName
+                          -> [UnqualComponentName]
+                          -> LegacyExeDependency
+                          -> Maybe ExeDependency
+desugarBuildToolSimple pname exeNames led =
+  if foundLocal
+  then Just $ ExeDependency pname toolName reqVer
+  else Map.lookup name whiteMap
+  where
+    LegacyExeDependency name reqVer = led
+    toolName = mkUnqualComponentName name
+    foundLocal = toolName `elem` exeNames
+    whitelist = [ "hscolour", "haddock", "happy", "alex", "hsc2hs", "c2hs"
```

Comment:
I prefer to avoid "whitelist"/"blacklist" these days. Instead I tend to use "allowlist"/"blocklist". But I don't feel strong enough about it to require anyone else to do the same.

Reply:
Ugh, you didn't write it, you just moved it. Well, I see no harm in changing it along the lines I propose anyway.